src/contract.rs

```diff
@@ -273,66 +256,50 @@ mod tests {
             vec![
                 attr("action", "liquid_stake"),
                 attr("native_amount", deposit_amount.to_string()),
-                attr("lst_mint_amount", expected_staked_amount.to_string()),
-                attr("exchange_rate", exchange_rate),
-                attr("denom", NATIVE_DENOM),
+                attr("native_ibc_denom", NATIVE_IBC_DENOM),
+                attr("native_base_denom", NATIVE_BASE_DENOM),
+                attr("ls_token_denom", LIQUIDSTAKE_DENOM),
                 attr("receiver", "receiver"),
             ]
         );
-
-        // ensure we can query the staked amount
-        let res = query(deps.as_ref(), mock_env(), QueryMsg::GetStakedLiquidity {}).unwrap();
-        let value: StakedLiquidityInfo = from_json(&res).unwrap();
-        assert_eq!(deposit_amount, value.staked_amount_native);
     }
 
     #[test]
-    fn liquid_stake_with_invalid_amount() {
+    fn handle_ls_reply_should_work() {
         let (mut deps, _env, _info) = default_instantiate();
 
-        let deposit_amount = Uint128::from(0u128);
-        let exchange_rate = "0.825537496882794638";
-
-        // Mock each pool in the querier
-        deps.querier
-            .mock_exchange_rate(MOCK_CHAIN_ID.to_string(), exchange_rate.to_string());
+        let msg = Reply {
+            id: 1,
+            result: cosmwasm_std::SubMsgResult::Ok(SubMsgResponse {
+                events: vec![],
+                data: Some(Binary::from_base64("Cv0BeyJjb250cmFjdCI6Indvcm1ob2xlMXl3NHd2MnpxZzl4a242N3p2cTNhenllMHQ4aDB4OWtneWczZDUzanltMjRneHQ0OXZkeXM2czhoN2EiLCJkZW5vbSI6bnVsbCwicmVjaXBpZW50Ijoic2VpMWRrZHdkdmtueDBxYXY1Y3A1a3c2OG1rbjNyOTltM3N2a3lqZnZrenR3aDk3ZHYybG0wa3NqNnhyYWsiLCJhbW91bnQiOiIxMDAwIiwicmVsYXllciI6InNlaTF2aGttMnF2Nzg0cnVseDh5bHJ1MHpwdnl2dzNtM2N5OXgzeHlmdiIsImZlZSI6IjAifQ==").unwrap())
```

Comment:
Why does this base64 data have wormhole and sei chain addresses?
Also, preferably don't put encoded data directly but put json as a raw Rust string and encode it and use it for easier test readability.

Reply:
Done.